contracts/serverless-v2/RelaySubscriptions.sol

```diff
@@ -0,0 +1,779 @@
+     * initializations of the logic contract.
+     * @param _relay The Relay contract.
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        Relay _relay
+    ) {
+        _disableInitializers();
+
+        if(address(_relay) == address(0)) 
+            revert RelaySubscriptionsInvalidRelay();
+        RELAY = _relay;
+    }
+
+    //-------------------------------- Overrides start --------------------------------//
+
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(ERC165Upgradeable, AccessControlUpgradeable) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    /// @inheritdoc UUPSUpgradeable
+    function _authorizeUpgrade(address /*account*/) internal view override onlyRole(DEFAULT_ADMIN_ROLE) {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    /// @notice Error for when zero address is provided for the admin.
+    error RelaySubscriptionsZeroAddressAdmin();
+
+    /**
+     * @notice Initializes the Relay contract with the specified admin and enclave images.
+     * @param _admin The address to be granted the DEFAULT_ADMIN_ROLE.
+     */
+    function initialize(address _admin) public initializer {
+        if (_admin == address(0)) revert RelaySubscriptionsZeroAddressAdmin();
+
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __UUPSUpgradeable_init_unchained();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+
+        jobSubsCount = (block.chainid << 192) | uint256(1) << 191;
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    Relay public immutable RELAY;
+
+    bytes32 private constant DOMAIN_SEPARATOR =
+        keccak256(
+            abi.encode(
+                keccak256("EIP712Domain(string name,string version)"),
+                keccak256("marlin.oyster.RelaySubscriptions"),
+                keccak256("1")
+            )
+        );
+
+    //-------------------------------- Job Subscription Start ----------------------------------//
+
+    struct Job {
+        uint256 startTime;
+        uint256 maxGasPrice;
+        uint256 usdcDeposit;
+        uint256 callbackDeposit;
+        uint256 callbackGasLimit;
+        address jobOwner;
+        address callbackContract;
+        bytes32 codehash;
+        bytes codeInputs;
+    }
+
+    struct JobSubscription {
+        uint256 periodicGap;
+        uint256 terminationTimestamp;
+        uint256 currentRuns;
+        uint256 lastRunTimestamp;
+        uint256 userTimeout;
+        address refundAccount;
+        Job job;
```

Comment:
can we merge these two structs, or use the struct definition from the Relay Contract (if possible). because, I don't see any benefit of these 2 separate structs.

Reply:
updated to be reused from Relay contract